mindmeld/active_learning/classifiers.py

```diff
@@ -531,3 +647,118 @@ def _update_eval_stats_intent_level(
+        """
+        er_eval_test_dict = {}
+        unsampled_idx_preds_pairs = {}
+        for domain, intents in domain_to_intents.items():
+            for intent in intents:
+                # Filter Queries
+                (
+                    _,
+                    filtered_sampled_queries,
+                ) = DataBucket.filter_queries_by_nlp_component(
+                    query_list=sampled_queries,
+                    component_type="intent",
+                    component_name=intent,
+                )
+                (
+                    filtered_unsampled_queries_indices,
+                    filtered_unsampled_queries,
+                ) = DataBucket.filter_queries_by_nlp_component(
+                    query_list=unsampled_queries,
+                    component_type="intent",
+                    component_name=intent,
+                )
+                _, filtered_test_queries = DataBucket.filter_queries_by_nlp_component(
+                    query_list=test_queries,
+                    component_type="intent",
+                    component_name=intent,
+                )
+                # Train
+                er = self.nlp.domains[domain].intents[intent].entity_recognizer
+                er.fit(queries=filtered_sampled_queries)
+
+                # Evaluate Test Queries
+                er_eval_test = er.evaluate(queries=filtered_test_queries)
+                er_eval_test_dict[f"{domain}.{intent}"] = er_eval_test
+
+                # Get Probability Vectors
+                er_queries_prob_vectors = MindMeldALClassifier._get_probs(
+                    classifier=er,
+                    queries=filtered_unsampled_queries,
+                    nlp_component_to_id=entity2id,
+                    nlp_component_type="entity",
+                )
+
+                for i, index in enumerate(filtered_unsampled_queries_indices):
+                    unsampled_idx_preds_pairs[index] = er_queries_prob_vectors[i]
+
+        indices = list(unsampled_idx_preds_pairs.keys())
+        indices.sort()
+        er_queries_prob_vectors = [
+            unsampled_idx_preds_pairs[index] for index in indices
+        ]
+        return er_queries_prob_vectors, er_eval_test_dict
+
+    def _update_eval_stats_entity_level(
+        self,
+        eval_stats: defaultdict,
+        er_eval_test_dict: Dict,
+        verbose: bool = False,
+    ):
+        """Update the eval_stats dictionary with evaluation metrics from entity
+        recognizers.
+
+        Args:
+            eval_stats (defaultdict): Evaluation metrics to be included in accuracies.json.
+            er_eval_test_dict (Dict): Dictionary mapping a domain.intent (str) to the
+                associated er_eval_test object.
+        """
+        for domain_intent, er_eval_test in er_eval_test_dict.items():
+            domain, intent = domain_intent.split(".")
+
+            if er_eval_test:
+                if domain not in eval_stats["accuracies"]:
```

Comment:
Let's move these two if checks into the plotting file itself

Reply:
I checked this, without these checks the line below can't populate this dictionary with the aggregate stats. We would need these or an equivalent here.